File: prime.js
var N = 100

range = function(start, end) {
  ary = [];
  for (i=start; i<=end; i++) { ary[i-start] = i }
  return ary;
};

primes = range(2, Math.sqrt(N)).reduce(function(nums, sieve){
  return nums.filter(function(num) {
    return (num % sieve) || (num == sieve);
  });
}, range(2, N))

print(primes);

File: prime.py
N = 100

import math
from functools import partial

primes = range(2, N+1)

for sieve in range(2, int(math.sqrt(N)) + 1):
  primes = filter(lambda num: (num % sieve != 0) or (num == sieve), primes)

print primes
